File: cmeRegexBench/benchmark.py
import re
import timeit

reps = 10


def process(lines, metric_help, metric_type, metric_sample):
    for line in lines:
        if metric_help.match(line):
            pass
        elif metric_type.match(line):
            pass
        elif metric_sample.match(line):
            pass


if __name__ == '__main__':

    f = open("samples.metrics", "r")

    lines = f.read().splitlines()

    metric_help = re.compile('^(#\s+HELP\s+([^\s]+).*)$')
    metric_type = re.compile('^(#\s+TYPE\s+([^\s]+).*)$')
    metric_sample = re.compile('^([^\#][^\s|{]*)\s*({(.*)})?\s+([^\s]+)$')

    start = timeit.default_timer()

    for i in range(reps):
        process(lines, metric_help, metric_type, metric_sample)

    print((timeit.default_timer() - start) / reps)

File: cmeRegexBench/benchmark.pl

use strict;
use Benchmark qw(:all) ;

sub read_lines {
    my ($file) = @_;
 
    open my $in, "<", $file;
    local $/ = undef;
    my $content = <$in>;
    close $in;
    return split /\n/, $content;
}

my @lines = read_lines('samples.metrics');

my $help = qr/^(#\s+HELP\s+([^\s]+).*)$/;
my $type_metric = qr/^(#\s+TYPE\s+([^\s]+).*)$/;
my $sample = qr/^([^\#][^\s|{]*)\s*({(.*)})?\s+([^\s]+)$/;

sub process { 
	foreach my $line (@lines) {
	   if ($line =~ m/$help/) # ~m is the match operator
	   {

	   }
	   elsif ($line =~ m/$type_metric/)
	   {
	      
	   }
	   elsif ($line =~ m/$sample/)
	   {
	     
	   }
	}
}



my $result = undef;





$result = timethis (10, 'process');

print "total CPU = ", $result->cpu_a, "\n";








